### DISCUSSION TITLE
Meter2 for H3 / or integrtion T-series

### DISCUSSION THREAD

**Discussion by Kyterion** [nathanmarlor/foxess_modbus#577] [General]
Is there a possibility for the H3 series to read out a generic inverter or, for example, a T series that is connected to the Meter2 interface to read out the power. In the modbus protocol I saw that the adress 31051 exists as Meter2, but I don't find it as an entity. I'm an amateur and a beginner :-)

**Comment by FozzieUK** [comment]
Yes if meter2 is connected you should find it as sensor.ct2_meter but it will probably be disabled by default (to reduce unnecessary polling)

If you got to Settings, Devices & Services, Foxess - Modbus and click on the ‘1 Device’ at the bottom of the sensor list it will say ‘+xx entities not shown’ - click on the CT2 meter entity, click the cog in top right corner and slide ‘Enabled’ to on - click update.

After a short while you should now see the CT2 meter entity

**Comment by Kyterion** [comment]
Thank you for the information. However, I've checked the sensor list, and there's no "+xx entities not shown" option at the bottom. Hence, I'm unable to locate the CT2 meter entity to enable it. Is there an alternative method to activate it?
![Fox](https://github.com/nathanmarlor/foxess_modbus/assets/117268443/5d7b564a-1dc6-4727-b6be-e24d258b3ca5)

**Reply by FozzieUK** [reply]
Ahh sorry, I just checked the code and there is no entity added for ct2 on the H3.

Let me double check the H3 protocol document and come back to you

**Reply by FozzieUK** [reply]
I have checked the various protocol documents and for the H3 the registers for meter2 do not appear to be exposed over modbus - where did you see address 31051 listed ?

**Reply by Kyterion** [reply]
Found it here, but I'm not sure if this also applies to the H3:
[KH_KA.Modbus.V1.01.00.2023070801.for.EMS.pdf](https://github.com/nathanmarlor/foxess_modbus/files/14705702/KH_KA.Modbus.V1.01.00.2023070801.for.EMS.pdf)

**Reply by canton7** [reply]
The KH is quite different to the H3 I'm afraid

**Reply by FozzieUK** [reply]
This isn’t the perfect solution, but in the absence of registers for the 2nd meter - you can install the Fox openapi cloud integration which does expose meter2power as one of the entities. 

It’s only sampled every 5 minutes (the dataloggers update interval) but at least you will be able to track it and sum it reasonably accurately.

https://github.com/macxq/foxess-ha

**Reply by Kyterion** [reply]
Thank you very much, that will help for now until something happens
